src/main/java/com/cuellojuan/dao/impl/GenericDAOImpl.java

```diff
@@ -226,81 +423,157 @@ public E find(Object entity) throws SQLException, NoSuchFieldException, IllegalA
         conn = dataSource.getConnection();
 
         String sql = "SELECT * FROM #TABLA WHERE #ID";
-        sql = sql.replace("#TABLA",tabla);
-        sql = sql.replace("#ID", devuelveID(entity));
+        sql = sql.replace("#TABLA", tabla);
+
+        sql = sql.replace("#ID", ID_conIgual.concat(invocarGetID(entity).toString()));
 
         PreparedStatement ps = conn.prepareStatement(sql);
         ResultSet rs = ps.executeQuery();
 
-        Object objetoClase = retornaInstanciaDeLaClase(entity).newInstance();
+         Object objetoClase = retornaInstanciaDeLaClase(entity).newInstance();
+
+        String idDelEntity = ID_conIgual.concat(invocarGetID(entity).toString());
+
+        idDelEntity = idDelEntity.replace(ID_conIgual,"");
+
+        try {
+
+            Class clase = objetoClase.getClass();
+            rs.next();
+
+            for (int i = 1; i < rs.getMetaData().getColumnCount() + 1; i++) {
+
+                Field campoParaSetear = clase.getDeclaredField(rs.getMetaData().getColumnName(i).toLowerCase());
+
+                Object objetoParaInstanciar;
+                String nombrePaqueteYCLase = NOMBRE_PAQUETE_DE_CLASES;
+
+                int tipodatoRS = rs.getMetaData().getColumnType(i);
+
+                switch (tipodatoRS) {
+                    case 4:
+
+                        if (clase.getDeclaredField(rs.getMetaData().getColumnName(i).toLowerCase()).getType() == int.class) {
+                            devuelveMetodo(clase, rs, i, campoParaSetear).invoke(objetoClase, rs.getInt(campoParaSetear.getName().toLowerCase()));
+                        } else {
+                            nombrePaqueteYCLase = nombrePaqueteYCLase.concat(".").concat(campoParaSetear.getType().getSimpleName());
+
+                            objetoParaInstanciar = Class.forName(nombrePaqueteYCLase).newInstance();
+
+                            Class claseDelObjParaInstanciar = Class.forName(nombrePaqueteYCLase);
+
+                            Method setMetodoDelObjeto = claseDelObjParaInstanciar.getMethod(SETID, int.class);
+                            setMetodoDelObjeto.invoke(objetoParaInstanciar, rs.getInt(rs.getMetaData().getColumnName(i).toLowerCase()));
+
+                            // todo esto es por que si el obj. es null significa, que va a devolver un obj. instanciado pero sin datos.
+                            Method getMetodoDelObjeto = claseDelObjParaInstanciar.getMethod(GETID, null);
+                            int idDelObjeto = (int) getMetodoDelObjeto.invoke(objetoParaInstanciar, null);
+                            if (idDelObjeto == 0)
```

Comment:
Reformlute this if-else, it is redundant.

Reply:
resuelto gracias.